src/platform/utilities/api/index.js

```diff
@@ -1,8 +1,68 @@
 import * as Sentry from '@sentry/browser';
 
 import environment from '../environment';
+import ENVIRONMENTS from '../../../site/constants/environments';
 import localStorage from '../storage/localStorage';
 
+let sessionTimeoutMinutesLengthSSO = 0;
+const SSOeEndpoint = () => {
+  let environmentPrefix;
+  switch (ENVIRONMENTS.BUILDTYPE) {
+    case ENVIRONMENTS.VAGOVSTAGING:
+      environmentPrefix = 'sqa.';
+      break;
+    case ENVIRONMENTS.VAGOVDEV:
+      environmentPrefix = 'int.';
+      break;
+    default:
+      environmentPrefix = '';
+  }
+
+  return `https://${environmentPrefix}eauth.va.gov/keepalive`;
+};
+
+function checkAndUpdateSSOeSession() {
+  const sessionExpiration = localStorage.getItem('sessionExpirationSSO');
+
+  if (sessionExpiration) {
+    const minutesUntilExpiration =
+      (sessionExpiration.getTime() - Date.now()) / (1000 * 60);
+
+    if (minutesUntilExpiration / sessionTimeoutMinutesLengthSSO < 0.67) {
+      // We want to minimize external keepalive calls, so this functions
+      // as self-enforced rate limiting
+      return;
+    }
+  }
+
+  fetch(SSOeEndpoint(), {
+    method: 'GET',
+    credentials: 'include',
+    cache: 'no-store',
+  })
+    .then(res => {
+      const hasSSOsession = res.headers.get('session-alive');
+
+      if (hasSSOsession === 'true') {
+        const sessionTimeout = res.headers.get('session-timeout');
```

Comment:
Is this a constant that represents the max length of a session in SSOe?

Reply:
I would move the comment about `session-timeout` being in seconds before this line. That context would be helpful before `sessionTimeout` is used to determine `sessionTimeoutMInutesLengthSSO`.

Reply:
To answer your question, if the value in response.headers.get('session-timeout') is "0" then there is no session. Any positive value would mean that the user has an active session, with the value being the seconds left in the session.

I agree about moving the timeout.